src/pages/api/event/[requestId].ts

```diff
@@ -1,28 +1,55 @@
-import { FingerprintJsServerApiClient, isEventError } from '@fingerprintjs/fingerprintjs-pro-server-api';
+import { isEventError } from '@fingerprintjs/fingerprintjs-pro-server-api';
 import { NextApiRequest, NextApiResponse } from 'next';
-import { BACKEND_REGION, SERVER_API_KEY } from '../../../server/const';
+import { fingerprintJsApiClient } from '../../../server/fingerprint-api';
 
 export default async function getFingerprintEvent(req: NextApiRequest, res: NextApiResponse) {
   const { requestId } = req.query as { requestId: string };
-  const client = new FingerprintJsServerApiClient({
-    region: BACKEND_REGION,
-    apiKey: SERVER_API_KEY,
-  });
 
+  return await tryGetFingerprintEvent(res, requestId);
+}
+
+/**
+ * It's possible the data about the specific event is not propagated to the Server API yet. This function retries for Not Found (404) requests.
+ * */
+async function tryGetFingerprintEvent(
+  res: NextApiResponse,
+  requestId: string,
+  retryCount = 5,
+  retryDelay: number = 3000,
+) {
   try {
-    const eventResponse = await client.getEvent(requestId);
+    const eventResponse = await fingerprintJsApiClient.getEvent(requestId);
     res.status(200).json(eventResponse);
   } catch (error) {
-    console.log(error);
-    if (isEventError(error)) {
-      res.statusMessage = `${error.error.code} - ${error.error.message}`;
-      res.status(error.status).json({
-        message: error.error.message,
-        code: error.error.code,
-      });
+    console.error(error);
+
+    if (retryCount === 1) {
+      handleFinalRetry(res, error);
+      return;
+    }
+
+    // Retry only Not Found (404) requests.
+    if (error.status === 404) {
+      setTimeout(() => tryGetFingerprintEvent(res, requestId, retryCount - 1, retryDelay), retryDelay);
     } else {
-      res.statusMessage = `Something went wrong ${error}`;
-      res.status(500);
+      handleOtherError(res, error);
```

Comment:
I think you want to call `handleFinalRetry` here (maybe name it just `sendErrorResponse`). Right now if the request fails on some other `EventError` (wrong API key for example) you don't get the nice JSON in the response even if it's available on the server.

Reply:
Consider an alternative flow 

```js
    if (error.status === 404 && retryCount > 1) {
      setTimeout(() => tryGetFingerprintEvent(res, requestId, retryCount - 1, retryDelay), retryDelay);
    } else {
      sendErrorResponse(res, error);
    }
```